lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -389,6 +394,40 @@ export class NormalComponent<
    */
   doInitialSchematicComponentRender() {
     if (this.root?.schematicDisabled) return
+
+    const { db } = this.root!
+    const manualPlacement =
+      this.getSubcircuit()?._getSchematicManualPlacementForComponent(this)
+
+    if (
+      this.props.schX !== undefined &&
+      this.props.schY !== undefined &&
+      manualPlacement
+    ) {
+      if (!this.schematic_component_id) {
+        const schematic_component = db.schematic_component.insert({
+          center: this._getGlobalSchematicPositionBeforeLayout(),
+          size: this._getSchematicBoxDimensions()?.getSize() ?? {
+            width: 0,
+            height: 0,
+          },
+          source_component_id: this.source_component_id!,
+        })
+        this.schematic_component_id = schematic_component.schematic_component_id
+      }
```

Comment:
this would be a huge bug. You need to wait for the schematic component to be created by the other parts of this function, not insert a duplicate

```suggestion
```

Reply:
if there is no schematic component id after the other parts of this fn execute, there's no schematic component, so no need to insert the warning